src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -218,8 +219,12 @@ const useAppPolling = (googleProject, workspaceName) => {
         await Ajax(signal).Apps.list(googleProject, { creator: getUser().email, saturnWorkspaceName: workspaceName }) :
         []
       setApps(newApps)
-      const app = getCurrentApp(newApps)
-      reschedule((app && _.includes(app.status, ['PROVISIONING', 'PREDELETING'])) ? 10000 : 120000)
+      _.forOwn(tool => {
+        if (tool.appType) {
+          const app = getCurrentAppForType(tool.appType)(newApps)
+          reschedule((app && _.includes(app.status, ['PROVISIONING', 'PREDELETING'])) ? 10000 : 120000)
+        }
```

Comment:
I'm finding this hard to parse inline... could we extract it into a separate, named method to make the intention clearer? ... or maybe just add a comment?